#270 title extension

VolkerHBZ opened the pull request:
changed the *.gsps and added OAI Export of the new properties

VolkerHBZ committed:
new properties

VolkerHBZ committed:
GUI integration of new properties

VolkerHBZ committed:
crossReferenceTitle supports new Properties now, incl. test

VolkerHBZ committed:
changed JSON names from priceType -> type and price -> amount

VolkerHBZ committed:
Merge remote-tracking branch 'remotes/origin/g3_master' into #270_title_extension

# Conflicts:
#	server/gokbg3/grails-app/views/apptemplates/_journal.gsp
#	server/gokbg3/src/integration-test/groovy/gokbg3/rest/IdentifierTestSpec.groovy

VolkerHBZ committed:
merged with master

hornmo committed:
Merge branch 'g3_master' into #270_title_extension

hornmo committed:
fix test

VolkerHBZ committed:
OAI export of the new properties + test

VolkerHBZ committed:
OAI-Export + test

VolkerHBZ committed:
removing unnecessary config attribute from the input fields in the gsp files.

VolkerHBZ committed:
Merge remote-tracking branch 'origin/#270_title_extension' into #270_title_extension

hornmo commented on server/gokbg3/grails-app/controllers/org/gokb/IntegrationController.groovy:
```diff
@@ -1559,13 +1568,13 @@ class IntegrationController {
+                log.debug("Completed date publishedFrom ${titleObj.publishedFrom} -> ${pubFrom}")
+
+                title_changed |= ClassUtils.setDateIfPresent(pubFrom, title, 'publishedFrom')
+                title_changed |= ClassUtils.setDateIfPresent(pubTo, title, 'publishedTo')
```
Like the date props the new title String properties (`series` and `subjectArea`) should be updated around here as well. Instead of setDateIfPresent for that we would use

https://github.com/openlibraryenvironment/gokb/blob/0a2a234c16b61a5501bbcfda722c26ed67cef655/server/gokbg3/src/main/groovy/com/k_int/ClassUtils.groovy#L174

hornmo commented on server/gokbg3/src/integration-test/groovy/gokbg3/rest/IdentifierTestSpec.groovy:
```diff
@@ -44,6 +44,7 @@ class IdentifierTestSpec extends AbstractAuthSpec {
       accept('application/json')
     }
     then:
+//    resp.status == 200 // WRONG!
```
Can be removed?

VolkerHBZ replied:
sometimes, the test returns a 200 status for no obvious reason. I suspect some html session to be alive still in the server for being responsible to validate the user.

hornmo commented on server/gokbg3/src/integration-test/resources/journal_prices_test.json:
```diff
@@ -0,0 +1,67 @@
+            "startDate": "2010-01-31"
+         },
+         {
+            "typeype": "topup",
```
Typo `typeype -> type`

hornmo commented on server/gokbg3/src/integration-test/resources/journal_prices_test.json:
```diff
@@ -0,0 +1,67 @@
+      "continuingSeries": "",
+      "status": "Current",
+      "series": "Mystery Cloud",
+      "subjectGroup": "Fringe",
```
`subjectGroup -> subjectArea` for consistency

VolkerHBZ replied:
this is left from a previous stage of implementation, when that field was named subjectGroup...

Changed it.

hornmo commented on server/gokbg3/src/integration-test/resources/journal_prices_test.json:
```diff
@@ -0,0 +1,67 @@
+            "value": "150346"
+         }
+      ],
+      "monographEdition": "null",
```
Monograph fields may be omitted entirely for `type: "Serial"`. Otherwise `null` would probably be the desired value here instead of the String value.

VolkerHBZ committed:
making IntegrationControllerSpec tests runnable individually
patching series & subjectArea as part of CrossReferenceSingleTitle
repair return value of ClassUtils.setXXifPresent methods
